### PR TITLE
make the landing page use a default promo for all copy (DS only)

### PR DESCRIPTION
**Pull request by johnduffell**
## What are you doing in this PR?

This adds a default promo to the DS landing page, meaning that the headline and subpara can be edited in the promo code tool.

## Why are you doing this?

This was the number 1 thing that marketing want when I asked.
Also, it's 10% time, and I decided to do this.

## Screenshots
promo code tool
image
landing page
image

### CODE REVIEW

File: support-frontend/app/controllers/DigitalSubscriptionController.scala [hunk 1]
```diff
@@ -184,3 +173,28 @@ class DigitalSubscriptionController(
...
+    val promoCodes: List[PromoCode] = queryPromos ++ DefaultPromotions.DigitalSubscription.all
+
+    val country = (for {
+      countryGroup <- CountryGroup.byId(countryCode)
+      country <- countryGroup.countries.headOption
+    } yield country).getOrElse(UK)
+    val promoCode = queryPromos.headOption.getOrElse(DefaultPromotions.DigitalSubscription.landing)
+    val promotionCopy = ProductPromotionCopy(promotionServiceProvider.forUser(false), stage)
+      .getCopyForPromoCode(promoCode, DigitalPack, country)
+    val readerType = if (orderIsAGift) Gift else Direct
```
Comment (rupertbates) [inline support-frontend/app/controllers/DigitalSubscriptionController.scala:195]: I'm not sure it adds much putting the price stuff in here - it complicates the return type and then we immediately extract the tuple in the calling method. Why not just deal with the copy in here and leave prices outside?

**Review by rupertbates — APPROVED** [review]
With one suggestion 👍

File: support-frontend/app/controllers/DigitalSubscriptionController.scala (hunk 1 above)
Comment (johnduffell) [reply support-frontend/app/controllers/DigitalSubscriptionController.scala:195]: good question.  The above method is perilously close to the 50 lines limit so I was a bit enthusiastic.  However you are right that they are mostly independent so they shouldn't be mixed here either.
I was half hoping the extracted method could be used in Paper and Weekly in the next PR.
I'll rejig it in this PR along the lines you suggest.

**Comment by johnduffell** [comment]
working in CODE - works both with and without the promo code present (uses default copy if not present)
Alwyas uses the "DEV" promo code in CODE and will always use the "PROD" one in PROD.

**Pull request merged**